arangod/VocBase/vocbase.cpp

```diff
@@ -2022,6 +2097,93 @@ TRI_voc_rid_t TRI_StringToRid(char const* p, size_t len, bool& isOld, bool warn)
   return HybridLogicalClock::decodeTimeStamp(p, len);
 }
 
+
+VocbaseOptions arangodb::getVocbaseOptions(VPackSlice const& options) {
+  // Invalid options will be silently ignored. Default values will be used
+  // instead.
+  //
+  // This Function will be called twice - the second time we do not run in
+  // the risk of consulting the ClusterFeature, because defaults are provided
+  // during the first call.
+
+  std::uint32_t replicationFactor =  1;   // real default will be read
+  std::uint32_t minReplicationFactor = 1; // from ClusterFeature if possible
+  std::string sharding;
+
+  //  sanitize input for vocbase creation
+  //  sharding -- must be "", "flexible" or "single"
+  //  replicationFactor must be "satellite" or a natural number
+  //  minReplicationFactor must be or a natural number
+  {
+    auto shardingSlice = options.get(StaticStrings::Sharding);
+    if(! (shardingSlice.isString()  &&
+          (shardingSlice.compareString("") == 0 || shardingSlice.compareString("flexible") == 0 || shardingSlice.compareString("single") == 0)
+         )) {
+      shardingSlice = VPackSlice::noneSlice();
+    } else {
+      sharding = shardingSlice.copyString();
+    }
+  }
+
+  ClusterFeature* cluster = dynamic_cast<ClusterFeature*>(application_features::ApplicationServer::lookupFeature("Cluster"));
+  {
+    VPackSlice replicationSlice = options.get(StaticStrings::ReplicationFactor);
+    bool isSatellite = (replicationSlice.isString() && replicationSlice.compareString(StaticStrings::Satellite) == 0 );
+    bool isNumber = (replicationSlice.isNumber() && replicationSlice.getUInt() > 0 );
+    if(!isSatellite && !isNumber){
+      if(cluster) {
+        replicationFactor = cluster->defaultReplicationFactor();
+      } else {
+        LOG_TOPIC("eeeee", ERR, Logger::CLUSTER) << "Can not access ClusterFeature to determine database replicationFactor";
+      }
+    } else if (isSatellite) {
+      replicationFactor = 0;
+    } else if (isNumber) {
+      replicationFactor = static_cast<decltype(replicationFactor)>(replicationSlice.getUInt());
+    }
+#ifndef USE_ENTERPRISE
+    if(replicationFactor == 0) {
+      if(cluster) {
+        replicationFactor = cluster->defaultReplicationFactor();
```

Comment:
Here we have detected that we want to have a defaultReplicationFactor of "satellite". And now this value is overwritten with the default replicationFactor...?

Reply:
`satellite` is not valid for non enterprise collections, if not overwritten collection creation will fail.